### DISCUSSION TITLE
Proxy support removed for tweepy api v2 ( Using tweepy 4.0 + ) ?

### DISCUSSION THREAD

**Discussion by global-live** [tweepy/tweepy#1784] [Questions] [answered]
Hi!

I noticed there is no more proxy support when creating the Tweepy Client, are you planning to add this in the future or we should look for alternate ways?

Thanks

**Comment by subbyte** [comment]
Tested working with environment variable `HTTP_PROXY`, `HTTPS_PROXY`, `http_proxy`, and `https_proxy`. Basically this is handled by the dependent library `requests`, not `tweepy` directly. Note that the log entry in the proxy may appear after the entire `tweepy` session ends. And since `tweepy` uses `requests.session()`, it could only show one proxy entry on the log.

**Comment by Harmon758** [comment, marked as answer]
[`Client.session`](https://docs.tweepy.org/en/latest/client.html#tweepy.Client.session) is the [`requests.Session`](https://docs.python-requests.org/en/stable/api/#requests.Session) used to make requests to the API.

You can change [its `proxies` attribute](https://docs.python-requests.org/en/stable/api/#requests.Session.proxies) after initializing `Client` or you can even set `Client.session` to your own instance of `requests.Session`.

As @subbyte pointed out, you can also use environment variables, as [the Proxies section of Request's documentation](https://docs.python-requests.org/en/stable/user/advanced/#proxies) covers.

Support for Tweepy API v2 through `Client` did not exist before Tweepy v4, and none of this has changed since it was added in Tweepy v4.0.0.
